src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/GetCurrentAndNextWorkitemWatermark.sql

```diff
@@ -0,0 +1,54 @@
+﻿/***************************************************************************
+    Stored procedure for getting current and next watermark for a workitem
+****************************************************************************/
+--
+-- STORED PROCEDURE
+--     GetCurrentAndNextWorkitemWatermark
+--
+-- DESCRIPTION
+--     Gets the current and next watermark.
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The partition key.
+--     @workitemUid
+--         * The Workitem instance UID.
+--
+-- RETURN VALUE
+--     Recordset with the following columns
+--          * Watermark
+--          * ProposedWatermark
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.GetCurrentAndNextWorkitemWatermark
```

Comment:
Do we need a new sproc or should we include this info in GetWorkitemMetadata

Reply:
we already returning the current watermark, we just need to return proposed watermark if needed. Like we can have a boolean to control.